tigerpath/scraper/scrape_parse.py

```diff
@@ -32,21 +141,27 @@ def scrape_parse_semester(term_code):
     # ahead of time
     TERM_PREFIX = FEED_PREFIX + "?term=" + str(TERM_CODE)
     DEP_PREFIX = TERM_PREFIX + "&subject="
-
-    # for now hardwire the namespaces--too annoying
-    PTON_NAMESPACE = u'http://as.oit.princeton.edu/xml/courseofferings-1_4'
+    VERSION_PREFIX = "&vers=1.5"
 
     CURRENT_SEMESTER = ['']
 
     h = HTMLParser()
 
-    def get_text(key, object):
-        return h.unescape(raise_if_none(object.find(key), "key " + key + " does not exist").text)
+    def get_text(key, object, fail_ok=False):
+        found = object.find(key)
+        if fail_ok and (found is None or found.text is None):
```

Comment:
I think it's more readable like this
`if fail_ok:`
____`if(found is None or found.text is None):`
________`return found`
____`else:`
________`return h.unescape(found.text)`
`else:`
 ____`return ParseError("key " + key + " does not exist")`

Reply:
That doesn't work in the case of `fail_ok=False`

Reply:
oh wait i see, my bad.